### PR TITLE
luajit: add option to build in gc64 mode

### PR DESCRIPTION
**Pull request by rcombs**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

`brew audit` gives me `* devel version should not decrease (from 2.1 to 2.1.0-beta3)`; not sure what's going on there (I didn't touch anything relevant).

### CODE REVIEW

**Comment by ilovezfs** [comment]
@DomT4 any idea why this isn't upstream default?

**Comment by DomT4** [comment]
As far as I'm aware it's an unfinished feature on the 2.1 branch & doesn't exist at all on the stable 2.0.x branch.

**Comment by rcombs** [comment]
This being closed makes it seem like it's effectively finished: https://github.com/LuaJIT/LuaJIT/issues/25

**Comment by DomT4** [comment]
The mailing list is still pretty active in terms of bug reports with this feature, so I would presume it's not "done". Nonetheless if this is added as an option it should be limited to `devel` for now since it doesn't exist in `stable` at all.

**Comment by rcombs** [comment]
Ah. I have no idea how to do that.

**Comment by ilovezfs** [comment]
`devel do` and `head do`

File: Formula/luajit.rb [hunk 1]
```diff
@@ -16,6 +14,12 @@ class Luajit < Formula
   devel do
     url "https://luajit.org/download/LuaJIT-2.1.0-beta3.tar.gz"
     sha256 "1ad2e34b111c802f9d0cdf019e986909123237a28c746b21295b63c9e785d9c3"
+    option "with-gc64", "Build with 64-bit support"
```
Comment (DomT4) [inline Formula/luajit.rb]: New line between `sha256` and the first `option` here & inside `head do`.

File: Formula/luajit.rb [hunk 2]
```diff
@@ -35,7 +39,12 @@ def install
     ENV.O2 # Respect the developer's choice.
 
     args = %W[PREFIX=#{prefix}]
-    args << "XCFLAGS=-DLUAJIT_ENABLE_LUA52COMPAT" if build.with? "52compat"
+
+    cflags = ""
+    cflags += "-DLUAJIT_ENABLE_LUA52COMPAT " if build.with? "52compat"
+    cflags += "-DLUAJIT_ENABLE_GC64" if build.with? "gc64"
```
Comment (DomT4) [inline Formula/luajit.rb]: Theoretically this should be something like:
```ruby
cflags << "-DLUAJIT_ENABLE_GC64" if !build.stable && build.with?("gc64")
```

File: Formula/luajit.rb [hunk 3]
```diff
@@ -35,7 +39,12 @@ def install
...
+
+    args << "XCFLAGS=#{cflags}" if cflags != ""
```
Comment (DomT4) [inline Formula/luajit.rb]: I'm not sure I love this syntax choice but I'll leave that down to ILZ. If we're going to use it though something like `unless cflags.zero?` should work & looks nicer.

File: Formula/luajit.rb [hunk 4]
```diff
@@ -35,7 +39,12 @@ def install
     ENV.O2 # Respect the developer's choice.
 
     args = %W[PREFIX=#{prefix}]
-    args << "XCFLAGS=-DLUAJIT_ENABLE_LUA52COMPAT" if build.with? "52compat"
+
+    cflags = ""
```
Comment (DomT4) [inline Formula/luajit.rb]: Probably `[]` rather than `""`.

**Comment by ilovezfs** [comment]
Thanks @rcombs!

**Pull request merged**